### PR TITLE
feat: introduce ToolchainCluster CRD that will replace KubeFedCluster

### PR DESCRIPTION
**Pull request by MatousJobanek**
## Description
This PR only introduces ToolchainCluster CRD that will replace the usage of KubeFedCluster CRD in the next PRs

## Checks
1. Have you run `make generate` target? **[yes/no]**

**yes**

2. Does `make generate` change anything in other projects (host-operator, member-operator, toolchain-common)? **[yes/no]** 
(NOTE: You can ignore it if the only changes are in the annotation `alm-examples` of the `ClusterServiceVersion` object)

**yes**

3. In case other projects are changed, please provides PR links.
    - host-operator: https://github.com/codeready-toolchain/host-operator/pull/256
    - member-operator: https://github.com/codeready-toolchain/member-operator/pull/184

### CODE REVIEW

File: pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
+package v1alpha1
+
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (alexeykazakov) [inline pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go:4]: I guess we should remove this.

File: pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ToolchainClusterConditionType string
+
+// These are valid conditions of a cluster.
+const (
+	// ClusterReady means the cluster is ready to accept workloads.
+	ToolchainClusterReady ToolchainClusterConditionType = "Ready"
```
Comment (tinakurian) [inline pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go:31]: this is minor; i think the standard for golang comments here is to start the comment with "ToolchainClusterReady..."

File: pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	// ClusterOffline means the cluster is temporarily down or not reachable
+	ToolchainClusterOffline ToolchainClusterConditionType = "Offline"
```
Comment (tinakurian) [inline pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go:33]: same here

File: pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+import (
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ToolchainClusterConditionType string
+
+// These are valid conditions of a cluster.
+const (
+	// ClusterReady means the cluster is ready to accept workloads.
```
Comment (tinakurian) [inline pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go:30]:
```suggestion
	// ToolchainClusterReady means the cluster is ready to accept workloads.
```

File: pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+	ToolchainClusterReady ToolchainClusterConditionType = "Ready"
+	// ClusterOffline means the cluster is temporarily down or not reachable
```
Comment (tinakurian) [inline pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go:32]:
```suggestion
	// ToolchainClusterOffline means the cluster is temporarily down or not reachable
```

File: pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go (hunk 4 above)
Comment (MatousJobanek) [reply pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go:30]: wrong copy-paste :grimacing: 
good catch, thanks

File: pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go (hunk 1 above)
Comment (MatousJobanek) [reply pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go]: you are right. I'm always confused when I have to add a copy of the licence clause and when not.
removed c2f020d

Comment (alexeykazakov) [reply pkg/apis/toolchain/v1alpha1/toolchaincluster_types.go]: Keeping the licence itself is fine. I was referring to the Kube Copyright actually. But it's fine now. Our repo does explicitly include the licence. So, I guess we don't have to duplicate it here anyway.

**Pull request merged**